bapsflib/_hdf/maps/controls/nixyz.py

```diff
@@ -0,0 +1,206 @@
+#
+# Copyright 2017-2019 Erik T. Everson and contributors
+#
+# License: Standard 3-clause BSD; see "LICENSES/LICENSE.txt" for full
+#   license terms and contributor agreement.
+#
+import h5py
+import numpy as np
+
+from bapsflib.utils.errors import HDFMappingError
+from warnings import warn
+
+from .contype import ConType
+from .templates import HDFMapControlTemplate
+
+
+class HDFMapControlNIXYZ(HDFMapControlTemplate):
+    """
+    Mapping module for control device 'NI_XYZ'.
+
+    Simple group structure looks like:
+
+    .. code-block:: none
+
+        +-- NI_XYZ
+        |   +-- <motion list name 1>
+        |   |   +--
+        .
+        .
+        .
+        |   +-- <motion list name N>
+        |   |   +--
+        |   +-- Run time list
+    """
+    def __init__(self, group: h5py.Group):
+        """
+        :param group: the HDF5 control device group
+        """
+        HDFMapControlTemplate.__init__(self, group)
+
+        # define control type
+        self._info['contype'] = ConType.motion
+
+        # populate self.configs
+        self._build_configs()
+
+    def _build_configs(self):
+        """Build the :attr:`configs` dictionary"""
+        # Assumptions:
+        # 1. only one NI_XYZ drive was ever built, so there will always
+        #    be only one configuration
+        #    - naming configuration 'config01'
+        # 2. there's only one dataset ever created 'Run time list'
+        # 3. there can be multiple motion lists defined
+        #    - each sub-group is a configuration for a different
+        #      motion list
+        #    - the name of the sub-group is the name of the motion list
+        #
+        # initialize configuration
+        cname = 'config01'
+        self.configs[cname] = {}
+
+        # check there are existing motion lists
+        if len(self.subgroup_names) == 0:
+            warn(self.info['group path']
+                 + ': no defining motion list groups exist')
+
+        # get dataset
+        try:
+            dset = self.group[self.construct_dataset_name()]
+        except KeyError:
+            why = ("Dataset '" + self.construct_dataset_name()
+                   + "' not found")
+            raise HDFMappingError(self.info['group path'], why=why)
+
+        # ---- define general config values                         ----
+        # none exist
+
+        # ---- define motion list values                            ----
+        self.configs[cname]['motion lists'] = {}
+
+        # get sub-group names (i.e. ml names)
+        _ml_names = []
+        for name in self.group:
+            if isinstance(self.group[name], h5py.Group):
+                _ml_names.append(name)
+
+        # a motion list group must have the attributes
+        # Nx, Ny, Nz, dx, dy, dz, x0, y0, z0
+        names_to_remove = []
+        for name in _ml_names:
+            if all(attr not in self.group[name].attrs
+                   for attr in ('Nx', 'Ny', 'Nz', 'dx', 'dy', 'dz', 'x0', 'y0', 'z0')):
```

Comment:
PEP8 line continuation

```suggestion
                   for attr in ('Nx', 'Ny', 'Nz', 'dx', 'dy', 'dz',
                                'x0', 'y0', 'z0')):
```